### ISSUE TITLE
[GAIA] The results of 'g.V(xxx).out()'  and 'g.V(xxx).out().values("vertex_id")' is inconsistent on Groot storage

### ISSUE THREAD

**Issue by lidongze0629** [alibaba/GraphScope#2359]
**Is your feature request related to a problem? Please describe.**

**Graph Schema**

Vertex label: company, size: 310 million
| 属性名 | 类型 | 主键 |
| -- | -- | -- |
| onecomp_id | STRING | False |
| ent_name | STRING | False  |
| vertex_id | STRING | **True** |

Edge label: edge, size: 10M
| 属性名 | 类型 | 主键 |
| -- | -- | -- |
| weight | DOUBLE | False |
| edge_id | STRING | False  |

**The number of results of `g.V(xxx).out()`  and `g.V(xxx).out().values("vertex_id")` is inconsistent(7 vs 5) on Groot storage**

```gremlin
gremlin> g.V(-1756735225279499108).out()
==>v[2509067887805302957]
==>v[-3811659230356509175].    # not exist in out().values()
==>v[-1641246016715921969]
==>v[-1547698570446649215]
==>v[-1077595549216581472]
==>v[-798875315838991321].   # not exist in out().values()
==>v[-741701238719865971]

gremlin> g.V(-1756735225279499108).out().values("vertex_id")
==>1:1201126000078824533
==>1:1191230000061212369
==>1:1200713000019841737
==>1:1181101000456742327
==>1:1200510000011765588

gremlin> g.V(2509067887805302957).valueMap()
==>{vertex_id=[1:1201126000078824533], onecomp_id=[1201126000078824533], ent_name=[xxxx]}
gremlin> g.V(-3811659230356509175).valueMap(). # not exist
==>{vertex_id=[1:1180716023542054464], onecomp_id=[1180716023542054464], ent_name=[xxxxx]}
gremlin> g.V(-741701238719865971).valueMap()
==>{vertex_id=[1:1200510000011765588], onecomp_id=[1200510000011765588], ent_name=[xxxx]}
gremlin> g.V(-798875315838991321).valueMap(). # not exist
==>{vertex_id=[1:1210313000047368737], onecomp_id=[1210313000047368737], ent_name=[xxxx]}
gremlin> g.V(-1077595549216581472).valueMap()
==>{vertex_id=[1:1181101000456742327], onecomp_id=[1181101000456742327], ent_name=[xxxx]}
gremlin> g.V(-1547698570446649215).valueMap()
==>{vertex_id=[1:1200713000019841737], onecomp_id=[1200713000019841737], ent_name=[xxxxx]}
gremlin> g.V(-1641246016715921969).valueMap()
==>{vertex_id=[1:1191230000061212369], onecomp_id=[1191230000061212369], ent_name=[xxxx]}
```

**Comment by BingqingLyu** [comment]
The reason is that when start up gie on groot, you may need to set extra settings correctly, e.g., 
`helm install company-test graphscope/graphscope-store --set extraConfig="pegasus.worker.num=8;pegasus.timeout=100"`  etc.

Otherwise, default value will be taken here: https://github.com/alibaba/GraphScope/blob/main/interactive_engine/compiler/src/main/java/com/alibaba/graphscope/common/config/PegasusConfig.java

In this case, the settings are given correctly. Thus the engine assumes that worker number and server number are 1s, and there's no need to shuffle the data, which occurs the above bug (i.e., without shuffle, we cannot find the data if it is not on current partition).

**Comment by siyuan0322** [comment]
Fixed by #2372
